astropy/time/core.py

```diff
@@ -1879,19 +1895,22 @@ def sidereal_time(self, kind, longitude=None, model=None):
 
         model_kwargs = available_models[model.upper()]
 
-        if isinstance(longitude, str) and longitude in ('tio', 'greenwich'):
-            longitude = 0
-            model_kwargs = model_kwargs.copy()
-            model_kwargs['include_tio'] = False
+        if isinstance(longitude, str):
+            if longitude in ('tio', 'greenwich'):
+                longitude = 0
+                model_kwargs = model_kwargs.copy()
+                model_kwargs['include_tio'] = False
+            else:
+                raise ValueError("Invalid longitude string")
```

Comment:
Actually, this is not correct, we should revert this. As is, it is possible and OK to pass in `longitude='12d'`. I'll do that...

Reply:
OK, I pushed a modified commit that does not change how string is handled, and added a test to ensure we do not remove it by mistake in the future!